### PR TITLE
Removed Pickle

### PR DESCRIPTION
**Pull request by nvidianz**
Implemented FOBS and replaced Pickle with it.

### CODE REVIEW

File: nvflare/apis/shareable.py [hunk 1]
```diff
@@ -12,8 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import pickle
-
+from ..fuel.utils import fobs
```
Comment (YuanTingHsieh) [inline nvflare/apis/shareable.py:15]: minor stuff: we might want to use absolute import for fobs

File: test/unit_test/fuel/test_fobs.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+    @classmethod
+    def setup_class(cls):
+
+        simple = Simple(123, "Simple Object", datetime.now())
+        dxo = DXO(DataKind.MODEL, {"object": simple}, {"tuple": ("test metadata", 456)})
+        cls.flare_obj = FlareObject("Test Flare Object", dxo)
+
+        cls.original_count = fobs.num_decomposers()
+        fobs.register(SimpleDecomposer())
```
Comment (YuanTingHsieh) [inline test/unit_test/fuel/test_fobs.py:78]: what if we don't provide the SimpleDecomposer?

File: nvflare/fuel/utils/fobs/README.rst [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+Usage
+-----
+
+FOBS defines following 4 functions, similar to Pickle,
+
+* :code:`serialize(obj)`: Serializes obj and returns bytes
+* :code:`serialize_stream(obj, stream)`: Serializes obj and writes the result to stream
+* :code:`deserialize(data)`: Deserializes the data and returns an object
+* :code:`deserialize_stream(stream)`: Reads data from stream and deserializes it int an object
```
Comment (nvkevlu) [inline nvflare/fuel/utils/fobs/README.rst:27]: I think "int" -> "into" here

File: nvflare/apis/shareable.py (hunk 1 above)
Comment (nvidianz) [reply nvflare/apis/shareable.py:15]: Good idea. Relative and wildcard imports should be disallowed.

File: test/unit_test/fuel/test_fobs.py (hunk 2 above)
Comment (nvidianz) [reply test/unit_test/fuel/test_fobs.py:78]: fobs throws TypeError in that case.

File: nvflare/fuel/utils/fobs/README.rst (hunk 3 above)
Comment (nvidianz) [reply nvflare/fuel/utils/fobs/README.rst:27]: Thanks!

**Review by YuanTingHsieh — CHANGES_REQUESTED** [review]
Minor changes are needed

File: nvflare/fuel/utils/decomposers/ml_decomposers.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Decomposers for types from Machine Learning frameworks and libraries."""
+from abc import ABC
+from io import BytesIO
+from typing import Any
+
+import numpy as np
+import tensorflow as tf
```
Comment (yanchengnv) [inline nvflare/fuel/utils/decomposers/ml_decomposers.py:20]: These would make NVFLARE Core depend on np, tf, and torch.

Please move this file to somewhere in app_common.

File: nvflare/private/fed/utils/numproto.py [hunk 5]
```diff
@@ -73,6 +73,4 @@ def proto_to_bytes(nda_proto: NDArray) -> bytes:
     Returns:
         Returns bytes.
     """
-    nda_bytes = BytesIO(nda_proto.ndarray)
-
-    return nda_bytes.read()
+    return nda_proto.ndarray
```
Comment (yanchengnv) [inline nvflare/private/fed/utils/numproto.py:76]: Is this right? This func is supposed to return bytes, but you are now returning NDArray.

**Review by yanchengnv — CHANGES_REQUESTED** [review]

**Comment by YuanTingHsieh** [comment]
@nvidianz should we close this ?

**Pull request closed without merging**